modules/oe_whitelabel_paragraphs/tests/src/Functional/InstallTest.php

```diff
@@ -0,0 +1,209 @@
+          'uri' => 'mailto:example@com',
+          'link_type' => 'email',
+        ],
+      ],
+      'field_oe_social_media_see_more' => [
+        'title' => 'Other social networks',
+        'uri' => 'https://europa.eu/european-union/contact/social-networks_en',
+      ],
+    ];
+
+    $ids = [];
+    foreach ($paragraphs_data as $name => $paragraph_data) {
+      $paragraph = Paragraph::create($paragraph_data);
+      $paragraph->save();
+      $ids[$name] = $paragraph->id();
+    }
+
+    self::assertTrue($this->moduleHandler->moduleExists('oe_whitelabel_legacy_paragraphs_test'));
+    self::assertFalse($this->moduleHandler->moduleExists('oe_whitelabel_paragraphs'));
+    self::assertTrue($this->moduleInstaller->install(['oe_whitelabel_paragraphs']));
+    self::assertFalse($this->moduleHandler->moduleExists('oe_whitelabel_paragraphs'));
+    $this->reloadServices();
+    self::assertTrue($this->moduleHandler->moduleExists('oe_whitelabel_paragraphs'));
+
+    $expected_created = [
+      'oe_description_list' => [
+        'oe_w_orientation' => 'horizontal',
+      ],
+      'oe_facts_figures' => [
+        'oe_w_n_columns' => '3',
+      ],
+      'oe_links_block' => [
+        'oe_w_links_block_orientation' => 'vertical',
+        'oe_w_links_block_background' => 'gray',
+      ],
+      'oe_social_media_follow' => [
+        'oe_w_links_block_background' => 'gray',
+      ],
+    ];
+
+    $expected_deleted = [
+      'oe_description_list' => [
+        'oe_bt_orientation' => TRUE,
+      ],
+      'oe_facts_figures' => [
+        'oe_bt_n_columns' => TRUE,
+      ],
+      'oe_links_block' => [
+        'oe_bt_links_block_orientation' => TRUE,
+        'oe_bt_links_block_background' => TRUE,
+      ],
+      'oe_social_media_follow' => [
+        'oe_bt_links_block_background' => TRUE,
+      ],
+    ];
+
+    // Produce reports instead of many individual assertions. This is less
+    // simple in code, but produces more useful output on test failure.
+    $actual_updated = [];
+    $actual_deleted = [];
+    foreach ($ids as $name => $id) {
+      $updated_paragraph = Paragraph::load($id);
+      self::assertNotNull($updated_paragraph);
+      foreach ($expected_created[$name] as $field_name => $value) {
+        if (!$updated_paragraph->hasField($field_name)) {
+          // The expected field was not created.
+          // Omit this entry in $actual_updated, to cause a fail below.
+          continue;
+        }
+        // The expected field was created, but the value might be wrong.
+        $actual_updated[$name][$field_name] = $updated_paragraph->get($field_name)->value;
+      }
+      foreach ($expected_deleted[$name] as $field_name => $deleted) {
+        $actual_deleted[$name][$field_name] = !$updated_paragraph->hasField($field_name);
+      }
+    }
+
+    // Compare the reports to the expected values.
+    self::assertSame($expected_created, $actual_updated);
+    self::assertSame($expected_deleted, $actual_deleted);
```

Comment:
Tests look good, but can we add at least one revision on one of them, to test that part? One of them is enough.